kayak_core/src/on_layout.rs

```diff
@@ -0,0 +1,52 @@
+use crate::layout::LayoutEvent;
+use crate::KayakContextRef;
+use std::fmt::{Debug, Formatter};
+use std::sync::{Arc, RwLock};
+
+/// A container for a function that handles layout
+///
+/// This differs from a standard [`Handler`](crate::Handler) in that it's sent directly
+/// from the [`KayakContext`](crate::KayakContext) and gives the [`KayakContextRef`]
+/// as a parameter.
+#[derive(Clone)]
+pub struct OnLayout(
+    Arc<RwLock<dyn FnMut(&mut KayakContextRef, &mut LayoutEvent) + Send + Sync + 'static>>,
```

Comment:
Ever since you brought this up, I'm debating on whether we should keep `&mut LayoutEvent` over `&LayoutEvent`.

My initial concern was consistency with `OnEvent` and future methods that require mutable access (similar to `Event::stop_propagation()`). However, I think you're right and that it might be more intuitive to users if this is immutable, so they don't try to "edit the layout".

Those "future methods" could then use some sort of interior mutability if necessary. Thoughts on that?

Reply:
It's a hard issue. Each way is backward incompatible.

I'd go with the less newbie confusing option (`&LayoutEvent`).

Reply:
I'm on board. Let's do that!